hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/expression/datetime/ToCharIntegrationTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.expression.datetime;
+
+import com.hazelcast.jet.sql.impl.expression.ExpressionTestSupport;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.Temporal;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.hazelcast.sql.SqlColumnType.VARCHAR;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class ToCharIntegrationTest extends ExpressionTestSupport {
+    private static Map<Class<? extends Temporal>, DateTimeFormatter> literals;
+
+    @BeforeClass
+    public static void setup() {
+        literals = new HashMap<>();
+        literals.put(LocalDate.class, DateTimeFormatter.ofPattern("'DATE' ''yyyy-MM-dd''"));
+        literals.put(LocalTime.class, DateTimeFormatter.ofPattern("'TIME' ''HH:mm:ss''"));
+        literals.put(LocalDateTime.class, DateTimeFormatter.ofPattern("'TIMESTAMP' ''yyyy-MM-dd HH:mm:ss''"));
+        literals.put(OffsetDateTime.class,
+                DateTimeFormatter.ofPattern("'TIMESTAMP WITH TIME ZONE' ''yyyy-MM-dd HH:mm:ssx''"));
+    }
+
+    @Test
+    public void testTemporals() {
+        check(LocalDate.of(2022, 9, 26), "Day, Mon DDth, YYYY", "Monday, Sep 26th, 2022");
+        check(LocalDate.of(2022, 9, 26), "DD TMMonth YYYY", "tr-TR", "26 Eylül 2022");
+        check(LocalTime.of(14, 53, 34), "HH:MI:SS AM", "2:53:34 PM");
+        check(LocalDateTime.of(2022, 9, 26, 14, 53), "Day, Mon DDth, HH24:MI", "Monday, Sep 26th, 14:53");
+//        check(LocalDateTime.of(2022, 9, 26, 14, 53).atOffset(ZoneOffset.ofHours(3)),
+//                "Day, Mon DDth, HH24:MI UTCFMTZH", "Monday, Sep 26th, 14:53 UTC+03");
+    }
+
+    @Test
+    public void testNumbers() {
```

Comment:
This check fails: `check(0, "pl9", "+0");`
It's due to the lack of Serialized on all objects that can be a part of the formatter.

Reply:
We don't recognize lowercase sign patterns and leading/trailing zeros are only printed if their corresponding mask is `"0"`. Therefore, this will work `check(0, "PL0", "+0");`.

Reply:
No support for lower-case patterns isn't documented. Why should we diverge here?
The `"0"` in the number `0` isn't a leading zero, it shouldn't format to `""`.

Reply:
Just noticed PG has better specification for the `9` pattern:
>digit position (can be dropped if insignificant)

The sole `0` in `0` isn't insignificant.

Reply:
In the [table of patterns][1], only `TH` has a lowercase variant.

[1]: https://www.postgresql.org/docs/14/functions-formatting.html#FUNCTIONS-FORMATTING-NUMERIC-TABLE